scenes/user/IngameUser.gd

```diff
@@ -0,0 +1,25 @@
+extends Resource
+
+class_name IngameUser
+
+var score : float = 0
+
+func _on_good_note(note_index: int, timing_error: float):
+	# TODO : add score calculations here.
+	score += 1.0
+	changed.emit()
+
+func get_data() -> Dictionary:
+	return {
+		"score" : score,
+	}
+
+func parse_data(data : Dictionary):
+	var chngd := false
+	if data.has("score"):
+		var sc = data.get("score")
+		if (sc is float || sc is int) && score != sc:
```

Comment:
Mentioning it again because this line is a great example, prefer boolean words over symbols.

if (sc is float or sc is int) and score != sc: